Source/relay/TourGuide/Sets/Sneaks.ash

```diff
@@ -0,0 +1,363 @@
+    int necksNCsLeft = 4 - countFriarNCs(necks_known_ncs, $location[The Dark Neck of the Woods]) - cartoAdjustment;
+    int heartNCsLeft = 4 - countFriarNCs(heart_known_ncs, $location[The Dark Heart of the Woods]);
+    int elbowNCsLeft = 4 - countFriarNCs(elbow_known_ncs, $location[The Dark Elbow of the Woods]);
+
+    // Also correcting total NCs left here; we are using a "count" for this, and thus we only get 1
+    //   out of however many NCs actually are left in these zones, because it just counts the elements
+    //   in the list rather than the # prepending the element in the list.
+    if (necksNCsLeft > 0) {
+        sneak95.listAppend(`{necksNCsLeft} Dark Neck`);
+        totalNCsRemaining += necksNCsLeft-1;
+    }
+    if (heartNCsLeft > 0) {
+        sneak95.listAppend(`{heartNCsLeft} Dark Heart`);
+        totalNCsRemaining += heartNCsLeft-1;
+    }
+    if (elbowNCsLeft > 0) {
+        sneak95.listAppend(`{elbowNCsLeft} Dark Elbow`);
+        totalNCsRemaining += elbowNCsLeft-1;
+    }
+    
+    // If the pref is any of these, you can still sneak the basement.
+    boolean [string] canSneakBasement = $strings[unstarted,started,step1,step2,step3,step4,step5,step6,step7];
+
+    if (canSneakBasement contains get_property("questL10Garbage")) sneak95.listAppend("1 castle basement");
+
+    // If the quest is unfinished, you can sneak theg top floor.
+    if (get_property("questL10Garbage")!= "finished") sneak95.listAppend("1 castle top floor");
+
+    tableLines[2] = populateSneakTable("95% Combat", sneak95);
+
+    totalNCsRemaining += sneak95.count();
+
+    // NEXT = 90% COMBAT SNEAKS
+    //    1x spookyraven bedroom
+    //    1x spookyraven bathroom
+
+    string [int] sneak90;
+
+    // I don't really think I need to import this but I'm tired and copy/pasting logic from the spookyraven tile seems fine.
+    QuestState dance_quest_state;
+    QuestStateParseMafiaQuestProperty(dance_quest_state, "questM21Dance");
+    if ($item[Lady Spookyraven's powder puff].available_amount() == 0 && dance_quest_state.mafia_internal_step < 4) sneak90.listAppend("1 spookyraven bathroom");
+    if ($item[Lady Spookyraven's dancing shoes].available_amount() == 0 && dance_quest_state.mafia_internal_step < 4) sneak90.listAppend("1 spookyraven gallery");
+
+    tableLines[3] = populateSneakTable("90% Combat", sneak90);
+
+    totalNCsRemaining += sneak90.count();
+
+    // Here is how Ezan built the lines into a table. It's kind of cute. First, make a placeholder "builder" line.
+    string [int] building_line;
+    foreach key in tableLines
+    {
+        // For each key, you append it to the empty building lines.
+        building_line.listAppend(tableLines[key]);
+
+        // However, if the key is even, you append to the table. This works, because you are appending a two-element
+        //   item into the table, so it creates a string [int] that creates a row in the table.
```

Comment:
I'm not gonna lie, this table logic is inscrutable to me. But like a lot of things in TourGuide, it is what it is. 🤷‍♂️

Reply:
yeah i added a ton of comments but i honestly don't think the comments help that much. it's just magic scripting.